### PR TITLE
Adds examining mouths with flashlights

### PR DESCRIPTION
**Pull request by QualityVan**
:cl: QualityVan
add: Point flashlights at mouths to see what's inside them
/:cl:

Because I got annoyed at not being able to tell if someone had a tongue unless they spoke.
Makes eye examination via flashlight work on carbons generally rather than just humans.
Also fixes a couple minor issues with examining eyes, including the ability to examine the eyes of someone who doesn't have a head and the ability for someone's pupils to narrow when they haven't got any.

### CODE REVIEW

**Comment by Metallusions** [comment]
will this work with penlights and flashlights or only one or the other?

**Comment by QualityVan** [comment]
Both. Penlights inherit from flashlights; their only unique attributes are size(being penshaped) and the ability to make a medical hologram. Penlights aren't quite as bright either.

**Comment by KorPhaeron** [comment]
Cool

**Comment by vcordie** [comment]
im sorry, medical hologram?

**Comment by QualityVan** [comment]
Yeah. Turn a penlight on and click on someone who isn't next to you, it'll make a thing.

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -48,28 +48,50 @@
...
+					else //they're okay!
+						to_chat(user, "<span class='notice'>[M]'s pupils narrow.</span>")
+
+			if("mouth")
+				if((M.head && M.head.flags_cover & HEADCOVERSMOUTH) || (M.wear_mask && M.wear_mask.flags_cover & MASKCOVERSMOUTH))
+					to_chat(user, "<span class='notice'>You're going to need to remove that [(M.head && M.head.flags_cover & HEADCOVERSMOUTH) ? "helmet" : "mask"] first.</span>")
+					return
+
+				if(M == user)
+					M.visible_message("[M] directs [src] to [M.p_their()] mouth.", "<span class='notice'>You point the flashlight into your mouth, but can't see anything.</span>")
```
Comment (ChangelingRain) [inline code/game/objects/items/devices/flashlight.dm]: if you're next to a mirror this should show you your tongue

**Comment by Core0verload** [comment]
How about making it detect other mouth organs and dental implants?

**Comment by QualityVan** [comment]
How many mouth organs are there besides breathing tubes? Vocal chords, I guess.

File: code/game/objects/items/devices/flashlight.dm [hunk 2]
```diff
@@ -79,18 +79,67 @@
...
+						var/obj/structure/mirror/mirror = locate(/obj/structure/mirror, user.loc)
+						if(mirror)
+							switch(user.dir)
+								if(1)
+									can_use_mirror = mirror.pixel_y > 0
+								if(2)
+									can_use_mirror = mirror.pixel_y < 0
+								if(4)
+									can_use_mirror = mirror.pixel_x > 0
+								if(8)
```
Comment (ChangelingRain) [inline code/game/objects/items/devices/flashlight.dm]: use the dir defines; in order, they're NORTH, SOUTH, EAST, and WEST.

File: code/game/objects/items/devices/flashlight.dm [hunk 3]
```diff
@@ -79,18 +79,67 @@
...
-										 "<span class='danger'>You direct [src] to [M]'s mouth.</span>")
-					var/obj/item/organ/tongue/T = M.getorganslot("tongue")
-					if(T)
-						to_chat(user, "<span class='notice'>Inside the mouth is \a [T.name].</span>")
+					user.visible_message("<span class='notice'>[user] directs [src] to [M]'s mouth.</span>",\
+										 "<span class='notice'>You direct [src] to [M]'s mouth.</span>")
+					if(mouth_organs.len)
+						to_chat(user, "<span class='notice'>Inside [their] mouth [mouth_organs.len > 1 ? "are" : "is"] [organ_list].</span>")
+					else
+						to_chat(user, "<span class='notice'>[M] doesn't have any organs in their mouth.</span>")
```
Comment (ChangelingRain) [inline code/game/objects/items/devices/flashlight.dm]: missed a their!

File: code/game/objects/items/devices/flashlight.dm [hunk 4]
```diff
@@ -79,18 +79,67 @@
...
 					return
 
+				var/their = M.p_their()
+
+				var/list/mouth_organs = new
+				for(var/obj/item/organ/O in M.internal_organs)
+					if(O.zone == "mouth")
+						mouth_organs.Add(O)
+				var/organ_list = ""
+				for(var/I = 1, I <= mouth_organs.len, I++)
```
Comment (ChangelingRain) [inline code/game/objects/items/devices/flashlight.dm]: for(var/I in 1 to mouth_organs.len)
also you need to check if(LAZYLEN(mouth_organs)) first so it doesn't fuck up if there are none

Comment (QualityVan) [inline code/game/objects/items/devices/flashlight.dm]: What's the benefit to I in range over I <= value, I++? The second won't hit an issue if value is zero in this case, and I figure len will be initialized to zero at line 89 so it being null isn't a concern

Comment (MrPerson) [inline code/game/objects/items/devices/flashlight.dm]: 1 in range is like twice as fast

**Comment by WJohn** [comment]
It should also show if they have a pill in one of their teeth.

**Pull request merged**